rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -91,6 +92,9 @@ typedef struct UseCountTopic
 
 typedef struct CustomParticipantInfo
 {
+  std::shared_ptr<eprosima::fastdds::dds::DomainParticipantFactory> factory_ =
+    eprosima::fastdds::dds::DomainParticipantFactory::get_shared_instance();
+
```

Comment:
> It should then avoid the automatic destruction of the factory (and the participants it created) until all the contexts in the process have been destroyed.

this will `eprosima::fastdds::dds::DomainParticipantFactory` object until all contexts in the process space are destroyed for sure. but i am not sure if this works for `eprosima::fastdds::dds::DomainParticipant`... does this mean `eprosima::fastdds::dds::DomainParticipantFactory` keeps the shared_ptr to `eprosima::fastdds::dds::DomainParticipant` until dtor?

Reply:
@MiguelCompany thanks for the info.

although i understand the code stack, i am not sure how / what this is trying to mitigate?

> This PR aims to mitigate them

can you elaborate this `them` issues in example if possible?

Reply:
In short: Any issue related with call stacks that include `__cxa_finalize()`

Recent examples:
* https://github.com/ros2/rmw_fastrtps/issues/761
* https://github.com/ros2/rmw_fastrtps/issues/760

Old (long standing) issues:
* https://github.com/ros2/rclcpp/issues/1766
* https://github.com/ros2/ros2/issues/1439

Reply:
Additionally, this was discussed with @ivanpauno [here](https://github.com/eProsima/Fast-DDS/discussions/2537) when the context concept was introduced